addons/hr_timesheet/models/hr_timesheet.py

```diff
@@ -53,13 +53,21 @@ def _compute_encoding_uom_id(self):
         for analytic_line in self:
             analytic_line.encoding_uom_id = analytic_line.company_id.timesheet_encode_uom_id
 
-    @api.depends('task_id', 'task_id.project_id', 'project_id')
-    def _compute_project_task_id(self):
+    @api.depends('task_id', 'task_id.project_id')
+    def _compute_project_id(self):
         for line in self.filtered(lambda line: not line.project_id):
             line.project_id = line.task_id.project_id
-        for line in self.filtered(lambda line: not line.project_id and line.project_id != line.task_id.project_id):
+
+    @api.depends('project_id')
+    def _compute_task_id(self):
+        for line in self.filtered(lambda line: not line.project_id):
             line.task_id = False
 
+    @api.onchange('project_id')
+    def _onchange_project_id(self):
+        if self.project_id != self.task_id.project_id:
+            self.task_id = False
+
```

Comment:
As we discussed, you said that it seems impossible to reset the task_id in the compute without breaking othe current behaviors. I would add a comment to explain why this should be like this maybe with a todo to check if it is still needed in the future. We should then avoid somebody to remove this apparently useless onchange and introduce a regression. :-)